detectionmetrics/utils/metrics.py

```diff
@@ -160,3 +160,51 @@ def get_accuracy(self) -> Tuple[np.ndarray, float]:
         )
         acc = np.sum(correct_per_class) / np.sum(total_per_class)
         return acc_per_class, acc
+
+    def get_precision(self) -> np.ndarray:
+        """Compute precision from confusion matrix. 
+        Precision = true positives/ predicted positives.
+
+        :return: per class precision
+        :rtype: np.ndarray
+        """
+        true_positives = np.diag(self.confusion_matrix)
+        predicted_positives = np.sum(self.confusion_matrix, axis=0)
```

Comment:
this gives the total number of samples, in order to get predicted positives you would have to subtract the predicted negatives. In other words, `predicted positives = true positives + false positives`